src/adaXT/base_model.pyx

```diff
@@ -13,9 +13,52 @@ from .leaf_builder.leaf_builder cimport (LeafBuilderClassification,
                                          LeafBuilderPartialQuadratic)
 
 import numpy as np
+from numpy.typing import ArrayLike
 
 
 class BaseModel:
+
+    def _check_max_features(
+        self, max_features: int | str | float | None
+    ) -> int | str | float | None:
+
+        if max_features is None:
+            return max_features
+        elif isinstance(max_features, int):
+            if max_features < 1:
+                raise ValueError("max_features can not be less than 1")
+            else:
+                return max_features
+        elif isinstance(max_features, float):
+            return max_features
+        elif isinstance(max_features, str):
+            if max_features == "sqrt":
+                return max_features
+            elif max_features == "log2":
+                return max_features
+            else:
+                raise ValueError("The only string options available for max_features are \"sqrt\", \"log2\"")
+        else:
+            raise ValueError("max_features can only be int, float, or in {\"sqrt\", \"log2\"}")
+
+    def _check_sample_weight(self, sample_weight: ArrayLike | None) -> np.ndarray:
+        if sample_weight is None:
+            return np.ones(self.X_n_rows, dtype=DOUBLE)
+        sample_weight = np.array(sample_weight, dtype=DOUBLE)
+        if sample_weight.shape[0] != self.X_n_rows:
+            raise ValueError("sample_weight should have as many elements as X and Y")
+        if sample_weight.ndim > 1:
+            raise ValueError("sample_weight has more than one dimension")
+        return sample_weight
+
+    def _check_sample_indices(self,
+                              sample_indices: ArrayLike | None
+                              ) -> np.ndarray:
+
+        if sample_indices is None:
+            return np.arange(0, self.X_n_rows, dtype=np.int32)
+        return np.array(sample_indices, dtype=np.int32)
+
```

Comment:
Is there a reason not to add the same checks as in _check_sample_weight here?

Reply:
Yes, sample_indices might be a subset of the total number of samples. For example when doing bootstrapping in the random forest, we might only want to have 5 of the total 1000 rows as part of the single decision tree. Even though we skip the check in randomforest this is just an example. Or when someone else is defining it, they might also wish for the functionality. However, the number of dimension check has been added.